base/src/com/google/idea/blaze/base/sync/projectstructure/ExternalModuleProvider.java

```diff
@@ -0,0 +1,14 @@
+package com.google.idea.blaze.base.sync.projectstructure;
+
+import com.intellij.openapi.extensions.ExtensionPointName;
+import com.intellij.openapi.module.Module;
+
+/**
+ * An EP provider for external modules.
+ */
+public interface ExternalModuleProvider {
+    ExtensionPointName<ExternalModuleProvider> EP_NAME =
+            ExtensionPointName.create("com.google.idea.blaze.base.sync.projectstructure.ExternalModuleProvider");
+
+    boolean isOwnedByExternalModule(Module module);
```

Comment:
it is not really `externalModule` that own this module but it is much better than `validation` so I'll approve (not that it matters :) )

Reply:
yeah, if it was an internal name I would change it afterwards, but this is an EP, so let's avoid changes once its merged. Could you please change that to `isOwnedByExternalPlugin` or `isOwnedByThirdPartyPlugin`?

Reply:
updated to 'isOwnedByExternalPlugin'. Please a take @tpasternak.